pkg/operator/controller/controller.go

```diff
@@ -0,0 +1,378 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/cluster-dns-operator/pkg/manifests"
+	operatorclient "github.com/openshift/cluster-dns-operator/pkg/operator/client"
+	"github.com/openshift/cluster-dns-operator/pkg/util/slice"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// DefaultDNSController is the name of the default DNS instance.
+	DefaultDNSController = "default"
+
+	// DNSControllerFinalizer is applied to an DNS before being considered for processing;
+	// this ensures the operator has a chance to handle all states.
+	DNSControllerFinalizer = "dns.openshift.io/dns-controller"
```

Comment:
Since we are changing this from `dns.openshift.io/default-cluster-dns`, can we change it to `operator.openshift.io/dns`?

Reply:
done